AppController/djinnServer.rb

```diff
@@ -145,11 +145,6 @@ def run_server(server)
   :SSLCertName => nil
 )
 
-trap('TERM') {
-  Djinn.log_debug("Received TERM signal: stopping AppController.")
-  server.shutdown
```

Comment:
What does sever.shutdown do? If we remove this code is the process not calling server.shutdown and possibly not shutting down ports cleanly?

Reply:
That was to shut down the SOAP server: after that call for example you will see that appscale status will error out since the controller is up but cannot be reached. No port are leftover, here is the output of netstat before and after stopping the controller (the process pid is 5635):

root@euca-10-10-6-12:~# netstat -alnp|grep -E '(5635|17443)'
tcp        0      0 0.0.0.0:17443           0.0.0.0:*               LISTEN      5635/ruby       
tcp        0      0 10.10.6.12:45305        10.10.6.12:2181         ESTABLISHED 5635/ruby       
tcp       32      0 10.10.6.12:39766        10.10.6.12:4343         CLOSE_WAIT  5635/ruby       
root@euca-10-10-6-12:~# netstat -alnp|grep -E '(5635|17443)'
root@euca-10-10-6-12:~#